SETUP.md

```diff
@@ -0,0 +1,160 @@
+     uint256 public constant DONATION_DENOMINATOR = 10000;
+
+
+```
+
+### generate
+
+Compiles [Solidity](https://soliditylang.org/) contracts and [TypeChain](https://github.com/ethereum-ts/Typechain) [TypeScript](https://www.typescriptlang.org/) bindings.
+```sh
+npm run generate
+```
+
+### deploy
+
+```
+npm run deploy -- \
+--key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 \
+--root 0xbfef011dd64abe7707cee7b3b74a00b86689c8451f548371073ce3c935e09984 \
+--numPubkeys 32
+```
+
+This should generate a `genesis.json` file that looks like:
+
+```json
+{
+    "parameters": {
+        "MAX_DEPTH": 32,
+        "MAX_DEPOSIT_SUBTREE_DEPTH": 2,
+        "STAKE_AMOUNT": "100000000000000000",
+        "BLOCKS_TO_FINALISE": 40320,
+        "MIN_GAS_LEFT": 10000,
+        "MAX_TXS_PER_COMMIT": 32,
+        "USE_BURN_AUCTION": true,
+        "DONATION_ADDRESS": "0x00000000000000000000000000000000000000d0",
+        "DONATION_NUMERATOR": 7500,
+        "GENESIS_STATE_ROOT": "0xbfef011dd64abe7707cee7b3b74a00b86689c8451f548371073ce3c935e09984"
+    },
+    "addresses": {
+        "frontendGeneric": "0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9",
+        "frontendTransfer": "0x5FC8d32690cc91D4c39d9d3abcBD16989F875707",
+        "frontendMassMigration": "0x0165878A594ca255338adfa4d48449f69242Eb8F",
+        "frontendCreate2Transfer": "0xa513E6E4b8f2a923D98304ec87F64353C4D5C853",
+        "blsAccountRegistry": "0x8A791620dd6260079BF849Dc5567aDC3F2FdC318",
+        "tokenRegistry": "0x610178dA211FEF7D417bC0e6FeD39F05609AD788",
+        "transfer": "0xA51c1fc2f0D1a1b8494Ed1FE312d7C3a78Ed91C0",
+        "massMigration": "0xB7f8BC63BbcaD18155201308C8f3540b07f84F5e",
+        "create2Transfer": "0x0DCd1Bf9A1b36cE34237eEaFef220932846BCD82",
+        "burnAuction": "0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6",
+        "exampleToken": "0x9A676e781A523b5d0C0e43731313A708CB607508",
+        "spokeRegistry": "0x9A9f2CCfdE556A7E9Ff0848998Aa4a0CFD8863AE",
+        "vault": "0x68B1D87F95878fE05B998F19b66F4baba5De1aed",
+        "depositManager": "0x3Aa5ebB10DC797CAC828524e59A333d0A371443c",
+        "rollup": "0xc6e7DF5E7b4f2A278906862b61205850344D4e7d",
+        "withdrawManager": "0x322813Fd9A801c5507c9de605d63CEA4f2CE6c44"
+    },
+    "auxiliary": {
+        "domain": "0x4ea7799478a7af2a47ba555f04aec4ae4ba240bf410d7c859c34c310f0413892",
+        "genesisEth1Block": 306,
+        "version": "20539ad4d99b3d3e4810de24c14ba41cdd89ea2c"
+    }
+}
+```
+
+The client will be parameterized using this file.
+
+## Run Client (Node)
+
+```sh
+npx ts-node ./scripts/simulate.ts --proposer
+```
+
+### repl (Hubble console)
+
+```sh
+npm run repl
+```
+
+Then in repl console, run a transfer:
+
+```sh
+hubble.transfer(0, 1, 1, 1)
+```
+
+## Run Tests
+
+Note: Full test suite run currently fails due to test state leak
+
+```sh
+npm run test
+```
+
+### fast/slow/client
+
+```sh
+npm run test -- test/fast/* # slow/client/etc.
+```
```

Comment:
@ChihChengLiang This is primarily based off setup instructions you gave me for onboarding, figured it would be useful for others as well :)